app/testOnly/controllers/SessionStorageServiceController.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testOnly.controllers
+
+import auth.{FrontendAuthorisedFunctions, MtdItUser}
+import config.{AgentItvcErrorHandler, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import play.api.Logger
+import play.api.mvc._
+import testOnly.models.SessionDataModel
+import testOnly.services.SessionDataService
+import uk.gov.hmrc.http.{HeaderCarrier, SessionId}
+import utils.AuthenticatorPredicate
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SessionStorageServiceController @Inject()(implicit val ec: ExecutionContext,
+                                                implicit override val mcc: MessagesControllerComponents,
+                                                val itvcErrorHandler: ItvcErrorHandler,
+                                                implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                val authorisedFunctions: FrontendAuthorisedFunctions,
+                                                val auth: AuthenticatorPredicate,
+                                                val sessionDataService: SessionDataService
+                                               ) extends ClientConfirmedController {
+
+  def show(): Action[AnyContent] = auth.authenticatedAction(isAgent = false) {
+    implicit user =>
+      handleShow(isAgent = false)
+  }
+
+  def showAgent: Action[AnyContent] = auth.authenticatedAction(isAgent = true) {
+    implicit mtdItUser =>
+      handleShow(isAgent = true)
+  }
+
+  private def handleShow(isAgent: Boolean)(implicit hc: HeaderCarrier, ec: ExecutionContext, user: MtdItUser[_]): Future[Result] = {
+    hc.sessionId match {
```

Comment:
I would suspect its fine to read sessionID from query parameters, there is not strict need to be attached to headerCarrier sessionId

Reply:
query params when accessing the page? if so you would need to know the session id when accessing the page, or create our own session id...

Reply:
I mean url query parameter can be used here, there is not strict need to rely on existing session for this POC

Reply:
then this would not be the actual sessionId though. it would be just an id that we call session id. i can do this though if you like

Reply:
my understanding of sessionId concept in this context that this is customised session across services to pass data from service A (data producer) to service B(data consumer), essentially internal sessionId for service A and/or service B has no direction relation to this concept, but again, this something we can raise as a question

Reply:
Who should we raise this with? We could call it sessionDataId perhaps

Reply:
this is POC, we can change things later when actual integration work would start